client/rpc/block.go

```diff
@@ -62,26 +67,61 @@ func getBlock(clientCtx client.Context, height *int64) ([]byte, error) {
 	// header -> BlockchainInfo
 	// header, tx -> Block
 	// results -> BlockResults
-	res, err := node.Block(context.Background(), height)
+	resBlock, err := node.Block(context.Background(), height)
 	if err != nil {
 		return nil, err
 	}
 
-	return legacy.Cdc.MarshalJSON(res)
+	out, err := mkBlockResult(resBlock)
+	if err != nil {
+		return out, err
+	}
+
+	return out, nil
 }
 
-// get the current blockchain height
-func GetChainHeight(clientCtx client.Context) (int64, error) {
+func GetBlockByHash(clientCtx client.Context, hashHexString string) (*cmt.Block, error) {
+	hash, err := hex.DecodeString(hashHexString)
+	if err != nil {
+		return nil, err
+	}
+
+	// get the node
 	node, err := clientCtx.GetNode()
 	if err != nil {
-		return -1, err
+		return nil, err
 	}
 
-	status, err := node.Status(context.Background())
+	resBlock, err := node.BlockByHash(context.Background(), hash)
+
 	if err != nil {
-		return -1, err
+		return nil, err
+	} else if resBlock.Block == nil {
+		return nil, fmt.Errorf("block not found with hash: %s", hashHexString)
 	}
 
-	height := status.SyncInfo.LatestBlockHeight
-	return height, nil
+	out, err := mkBlockResult(resBlock)
+	if err != nil {
+		return out, err
+	}
+
+	return out, nil
+}
+
+func mkBlockResult(resBlock *coretypes.ResultBlock) (*cmt.Block, error) {
+	return sdk.NewResponseResultBlock(resBlock, resBlock.Block.Time.Format(time.RFC3339)), nil
+}
```

Comment:
What's the point of this func? It never returns an error. You might as well just call `NewResponseResultBlock` directly.

Reply:
Yeah, took care of it - thanks